indico/modules/events/requests/controllers.py

```diff
@@ -189,11 +189,12 @@ def _check_access(self):
     def _process(self):
         if self.request.state not in {RequestState.pending, RequestState.accepted}:
             raise BadRequest
-        try:
```

Comment:
This doesn't look correct. the `else` below is now part of the `if` (doesn't make much sense, the if body raises an exception) and the "You have witdrawn" message is flashed even if there was a RequestModuleError.

The previous code looked fine; all that was missing was the `self.commit = False`..